cert/spec.go

```diff
@@ -343,19 +310,6 @@ func (spec *Spec) CheckDiskPKI() error {
 	}
 	metrics.KeysizeMismatchCount.WithLabelValues(spec.Path).Set(0)
 
-	// Check that certificate hostnames match spec hostnames
-	certData, err := ioutil.ReadFile(certPath)
-	if err != nil {
-		return err
-	}
-	p, _ := pem.Decode(certData)
-	if p == nil {
-		return errors.New("Unable to pem decode certificate on disk")
-	}
-	cert, err := x509.ParseCertificate(p.Bytes)
-	if err != nil {
-		return err
-	}
 	// confirm that pkix is the same.  This catches things like OU being changed; these are slices
 	// of slices and there isn't a usable equality check, thus the .String() usage.
 	if csrRequest.Name().String() != cert.Subject.String() {
```

Comment:
I'm worried this will cause unnecessary failures. The way crypto/x509/pkix constructs a Name String (https://golang.org/src/crypto/x509/pkix/pkix.go#L203) differs from how cfssl constructs it (https://github.com/cloudflare/cfssl/blob/master/csr/csr.go#L167). 

But also there's no pkix.Name equality function -- so I don't have any other alternatives to suggest really. Except to implement a method like `hostnameEquals` below to manually go through each `pkix.Name` field we care about and check for String equality there.

Reply:
As far as I can tell, the only issue that code arise here is if the CA signer injects in additional pkix fields- which it shouldn't, it should just use the CSR as presented.

If the signer does that mutation, it'll definitely cause issues.  Barring that- the raw data for the cert vs csr will align, thus the comparison logic *should* align.  In my testing/validation this is working, but I'm definitely open to anything that has less of a fragile smell to it.

Reply:
Yeah you're probably right. In the off chance this does cause issues hopefully the logging/alerting will be clear enough to debug